### PR TITLE
ALL: remove file extensions to support more than tga

### PR DESCRIPTION
**Pull request by mgerhardy**
Added png support to spraylogos and skins.

most of the code parts are just removing the extension anyway. But this change allows us to replace an existing and referenced tga image with e.g. a png version
    
See issue #35 and #63

### CODE REVIEW

File: code/q3_ui/ui_gameinfo.c [hunk 1]
```diff
@@ -839,9 +839,9 @@ void UI_SearchSpraylogos( void ) {
 		nameLen = strlen( namePtr );
 		
 		// ignore anything but jpg and tga
```
Comment (kai-li-wop) [inline code/q3_ui/ui_gameinfo.c]: Maybe to adjust the comment regarding png support.

**Comment by kai-li-wop** [comment]
I tried to run this build on Windows 10 to test the adjustments regarding PNG, however WoP can no longer find the renderer.

Trying to load "renderer_opengl1_x86_64.dll" from "C:\Users\Kai\Documents\World of Padman\WoP Test"...
Loading "renderer_opengl1_x86_64.dll" failed
failed:
"Failed loading C:\Users\Kai\Documents\World of Padman\WoP Test\renderer_opengl1_x86_64.dll: Das angegebene Modul wurde nicht gefunden."

However, the file is on the spot. To get WoP running again, it is sufficient to restore the renderer_opengl1_x86_64.dll of the previous build. WoP then starts normally and even with working PNG support in the player menu. Somehow the OpenGL 1 renderer of this build is faulty.

**Comment by mgerhardy** [comment]
@kai-li-wop Could you please recheck with the latest binaries? The dll file used malloc and free instead of the renderer interface functions to allocate memory. This should be fixed now.

File: code/q3_ui/ui_gameinfo.c (hunk 1 above)
Comment (mgerhardy) [reply code/q3_ui/ui_gameinfo.c]: this is already done differently in a following commit - but that is already based on the re-formatting (as posted on the discord server). Thus, this is going to be part of another pull request.

**Comment by kai-li-wop** [comment]
@mgerhardy: Sorry to say, but there is no change of the situation. As soon as I start WoP with the latest binaries and the new renderer-dll file, the same error message appears. As soon as I switch back to the renderer-dll of the [prior build](https://github.com/PadWorld-Entertainment/worldofpadman/actions/runs/771026158) but keeping the latest binaries, WoP runs just fine.

**Comment by mgerhardy** [comment]
https://github.com/PadWorld-Entertainment/worldofpadman/pull/48

**Comment by mgerhardy** [comment]
Tested on Windows (and OSX) now. Latest version works on windows, linux and osx.

**Comment by kai-li-wop** [comment]
I can approve, that the renderer is loaded now like expected. Unfortunately, there is some issue now with custom player models. Please have a look at #87.

**Comment by mgerhardy** [comment]
Thanks a lot - it would indeed help if you could send me one of the custom models that works fine and one that isn't anymore. Please also just post the issues here to the PR - it's easier to follow and it isn't an issue yet of the main branch - but only of this PR. Thanks for testing and checking it.

**Comment by mgerhardy** [comment]
Should be fixed now - there was an off-by-one error. Please recheck - the wolf is still missing an icon as far as i can tell.

File: code/q3_ui/ui_gameinfo.c (hunk 1 above)
Comment (mgerhardy) [reply code/q3_ui/ui_gameinfo.c]: done

**Comment by kai-li-wop** [comment]
Something got fixed, the wolf is shown OK now in the player menu. But I still have the problem, that, when I chose the Johnny5 model, which has png icons, it is not loaded correctly ingame. The game loads the Padman model instead. When I go back to player setup via the ingame menu while I am in an arena, the player menu is not showing the skin icons of the Johnny 5 model. It is not possible to select any Johnny 5 skin there. So the player menu is still broken when accessed from ingame. Please try to follow my steps.
1. Chose the Johnny 5 player model from player menu
2. Start an arena in ffa for example
3. Press U to see third person view, Padman is shown instead of Johnny 5
4. Press Esc to call the ingame menu and chose setup
5. Go to player menu and chose Johnny 5
6. No png icons are shown, so it is not possible to select a Johnny 5 skin

**Comment by kai-li-wop** [comment]
OK, everything works now like it should.

**Review by kai-li-wop — APPROVED** [review]
The PNG support is fully working now.

**Pull request merged**